### ISSUE TITLE
[boost-thread] Build error on x64-windows

### ISSUE THREAD

**Issue by gideonr1** [microsoft/vcpkg#39359]
Copy issue body from C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/issue_body.md
[issue_body.md](https://github.com/user-attachments/files/15886017/issue_body.md)

**Comment by gideonr1** [comment]
Package: boost-thread:x64-windows-static@1.77.0

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.40.33811.0
-    vcpkg-tool version: 2024-03-14-7d353e869753e5609a1f1a057df3db8fd356e49d
    vcpkg-readonly: true
    vcpkg-scripts version: 7f9f0e44db287e8e67c0e888141bfa200ab45121

**To Reproduce**

`vcpkg install `
**Failure logs**

```
-- Using cached boostorg-thread-boost-1.77.0.tar.gz.
-- Cleaning sources at C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/src/ost-1.77.0-7bf4191b6c.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/Gideon/AppData/Local/vcpkg/downloads/boostorg-thread-boost-1.77.0.tar.gz
-- Using source at C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/src/ost-1.77.0-7bf4191b6c.clean
-- Including C:/Users/Gideon/AppData/Local/vcpkg/registries/git-trees/1c097a032a997463e15245272e40ea9298554f8b/b2-options.cmake
-- Found external ninja('1.11.0').
-- Configuring x64-windows-static
-- Building x64-windows-static-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j9
    Working Directory: C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg
    See logs for more information:
      C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\install-x64-windows-static-dbg-out.log

Call Stack (most recent call first):
  C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:102 (vcpkg_execute_build_process)
  C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/boost-modular-build.cmake:105 (vcpkg_cmake_install)
  C:/Users/Gideon/AppData/Local/vcpkg/registries/git-trees/1c097a032a997463e15245272e40ea9298554f8b/portfile.cmake:15 (boost_modular_build)
  scripts/ports.cmake:175 (include)



```
<details><summary>C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\install-x64-windows-static-dbg-out.log</summary>

```
Change Dir: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg'

Run Build Command(s): "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v -v -j9 install
[1/2] C:\WINDOWS\system32\cmd.exe /C "cd /D C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean\build && C:\Code\OTC\otc4-grpc-server\vcpkg_installed\x64-windows\tools\boost-build\b2.exe toolset=msvc --user-config=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg/user-config.jam --stagedir=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg/stage --build-dir=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg --disable-icu runtime-link=static link=static address-model=64 architecture=x86 target-os=windows threadapi=win32 variant=debug /boost/thread//boost_thread --with-atomic --with-random --with-date_time --with-filesystem --with-system --with-thread --with-chrono -j8 -sBOOST_ROOT=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/tools/boost-build -sBOOST_BUILD_PATH=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/tools/boost-build --debug-configuration --debug-building --debug-generators --ignore-site-config --hash -q debug-symbols=on -d +2 threading=multi stage"
FAILED: CMakeFiles/boost C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg/CMakeFiles/boost 
C:\WINDOWS\system32\cmd.exe /C "cd /D C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean\build && C:\Code\OTC\otc4-grpc-server\vcpkg_installed\x64-windows\tools\boost-build\b2.exe toolset=msvc --user-config=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg/user-config.jam --stagedir=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg/stage --build-dir=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg --disable-icu runtime-link=static link=static address-model=64 architecture=x86 target-os=windows threadapi=win32 variant=debug /boost/thread//boost_thread --with-atomic --with-random --with-date_time --with-filesystem --with-system --with-thread --with-chrono -j8 -sBOOST_ROOT=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/tools/boost-build -sBOOST_BUILD_PATH=C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/tools/boost-build --debug-configuration --debug-building --debug-generators --ignore-site-config --hash -q debug-symbols=on -d +2 threading=multi stage"
notice: found boost-build.jam at C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/tools/boost-build/boost-build.jam
notice: loading B2 from C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/tools/boost-build/src/kernel/bootstrap.jam
notice: Site configuration files will be ignored due to the
notice: --ignore-site-config command-line option.
notice: Loading explicitly specified user configuration file:
    C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\x64-windows-static-dbg\user-config.jam
notice: Searching 'C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\x64-windows-static-dbg' for user-config configuration file 'user-config.jam'.
notice: Loading user-config configuration file 'user-config.jam' from 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/vcpkg/blds/boost-thread/x64-windows-static-dbg'.
C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\x64-windows-static-dbg\user-config.jam:8: Unescaped special character in argument <linkflags>/machine:x64
notice: will use 'C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe' for msvc, condition <toolset>msvc-144
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>x86/<address-model>32', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat x86'
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>/<address-model>32', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat x86'
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>x86/<address-model>64', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat amd64'
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>/<address-model>64', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat amd64'
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>x86/<address-model>', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat amd64'
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>/<address-model>', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat amd64'
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>ia64/<address-model>64', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat x86_ia64'
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>ia64/<address-model>', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat x86_ia64'
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>arm/<address-model>32', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat x86_arm'
notice: [msvc-cfg] condition: '<toolset>msvc-144/<architecture>arm/<address-model>64', setup: 'C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows/share/boost-build/nothing.bat x86_arm64'

Building target '../build/boost_thread'
     Build request: object(property-set)@373 <address-model>64 <architecture>x86 <debug-symbols>on <link>static <runtime-link>static <target-os>windows <threadapi>win32 <threading>multi <toolset>msvc <variant>debug
     Command line free features:
     Target requirements: <conditional>@Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean\build>%Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean\build>.requirements <define>BOOST_ALL_NO_LIB=1 <include>../include&&/C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows-static/include <library>/boost/date_time//boost_date_time <link>shared:<define>BOOST_THREAD_BUILD_DLL=1 <link>static:<define>BOOST_THREAD_BUILD_LIB=1 <relevant>cxxflags:<relevant>toolset <relevant>define:<relevant>link <relevant>define:<relevant>target-os <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <relevant>warnings:<relevant>toolset <tag>@Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean\build>%Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean\build>.tag <target-os>windows:<define>BOOST_USE_WINDOWS_H <target-os>windows:<define>WIN32_LEAN_AND_MEAN <threading>multi <toolset>clang-3.0:<cxxflags>-Wno-delete-non-virtual-dtor <toolset>clang:<cxxflags>-Wextra <toolset>clang:<cxxflags>-Wno-long-long <toolset>clang:<cxxflags>-Wno-unused-parameter <toolset>clang:<cxxflags>-Wno-variadic-macros <toolset>clang:<cxxflags>-Wunused-function <toolset>clang:<warnings>on <toolset>darwin-4.6.2:<cxxflags>-Wno-delete-non-virtual-dtor <toolset>darwin-4.7.0:<cxxflags>-Wno-delete-non-virtual-dtor <toolset>darwin-4:<cxxflags>-Wno-variadic-macros <toolset>darwin-5:<cxxflags>-Wno-variadic-macros <toolset>darwin:<cxxflags>-Wextra <toolset>darwin:<cxxflags>-Wno-long-long <toolset>darwin:<cxxflags>-Wno-unused-parameter <toolset>darwin:<cxxflags>-Wunused-function <toolset>darwin:<cxxflags>-fpermissive <toolset>darwin:<cxxflags>-pedantic <toolset>gcc-4:<cxxflags>-Wno-variadic-macros <toolset>gcc-5:<cxxflags>-Wno-variadic-macros <toolset>gcc:<cxxflags>-Wextra <toolset>gcc:<cxxflags>-Wno-long-long <toolset>gcc:<cxxflags>-Wno-unused-parameter <toolset>gcc:<cxxflags>-Wunused-function <toolset>gcc:<cxxflags>-pedantic <toolset>intel:<cxxflags>-wd1418 <toolset>intel:<cxxflags>-wd193,304,383,444 <toolset>intel:<cxxflags>-wd2415 <toolset>intel:<cxxflags>-wd593,981 <toolset>msvc,<runtime-link>shared:<threading>multi <toolset>msvc:<cxxflags>/wd4100 <toolset>msvc:<cxxflags>/wd4512 <toolset>msvc:<cxxflags>/wd6246 <toolset>pathscale:<cxxflags>-Wno-long-long <toolset>pathscale:<cxxflags>-pedantic <warnings>all

     Common properties:  <address-model>64 <architecture>x86 <asynch-exceptions>off <coverage>off <cxxflags>/wd4100 <cxxflags>/wd4512 <cxxflags>/wd6246 <debug-store>object <debug-symbols>on <define>BOOST_ALL_NO_LIB=1 <define>BOOST_THREAD_BUILD_LIB=1 <define>BOOST_THREAD_USES_CHRONO <define>BOOST_THREAD_WIN32 <define>BOOST_USE_WINDOWS_H <define>WIN32_LEAN_AND_MEAN <embed-manifest-via>linker <embed-manifest>on <exception-handling>on <extern-c-nothrow>off <hardcode-dll-paths>true <host-os>windows <include>../include&&/C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows-static/include <inlining>off <install-dependencies>off <library>/boost/chrono//boost_chrono <library>/boost/date_time//boost_date_time <link>static <midl-robust>yes <midl-stubless-proxy>yes <optimization>off <os>NT <pch>on <preserve-test-targets>on <profiling>off <relevant>cxxflags:<relevant>toolset <relevant>define:<relevant>link <relevant>define:<relevant>target-os <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <relevant>warnings:<relevant>toolset <response-file>auto <rtti>on <runtime-debugging>on <runtime-link>static <stdlib>native <strip>off <suppress-import-lib>false <symlink-location>project-relative <tag>@Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean\build>%Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean\build>.tag <target-os>windows <testing.execute>on <threadapi>win32 <threading>multi <toolset-msvc:version>144 <toolset>msvc <user-interface>console <variant>debug <vectorize>off <warnings-as-errors>off <warnings>all <windows-api>desktop

     Building target '../boost_chrono'
         Build request: object(property-set)@398 <address-model>64 <architecture>x86 <asynch-exceptions>off <coverage>off <debug-store>object <debug-symbols>on <embed-manifest-via>linker <embed-manifest>on <exception-handling>on <extern-c-nothrow>off <inlining>off <link>static <midl-robust>yes <midl-stubless-proxy>yes <optimization>off <os>NT <pch>on <preserve-test-targets>on <profiling>off <relevant>variant <rtti>on <runtime-debugging>on <runtime-link>static <stdlib>native <strip>off <target-os>windows <testing.execute>on <threadapi>win32 <threading>multi <toolset-msvc:version>144 <toolset>msvc <variant>debug <vectorize>off <warnings-as-errors>off <warnings>all <windows-api>desktop
         Command line free features:
         Target requirements: <define>BOOST_ALL_NO_LIB=1 <file>/boost//C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows-static/debug/lib/boost_chrono-vc140-mt-gd.lib <include>../include&&/C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows-static/include <library>/boost/date_time//boost_date_time <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <tag>@Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean>%Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean>.tag <toolset>msvc,<runtime-link>shared:<threading>multi <variant>debug

         Common properties:  <address-model>64 <architecture>x86 <asynch-exceptions>off <coverage>off <debug-store>object <debug-symbols>on <define>BOOST_ALL_NO_LIB=1 <embed-manifest-via>linker <embed-manifest>on <exception-handling>on <extern-c-nothrow>off <file>/boost//C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows-static/debug/lib/boost_chrono-vc140-mt-gd.lib <hardcode-dll-paths>true <host-os>windows <include>../include&&/C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows-static/include <inlining>off <install-dependencies>off <library>/boost/date_time//boost_date_time <link>static <midl-robust>yes <midl-stubless-proxy>yes <optimization>off <os>NT <pch>on <preserve-test-targets>on <profiling>off <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <relevant>variant <response-file>auto <rtti>on <runtime-debugging>on <runtime-link>static <stdlib>native <strip>off <suppress-import-lib>false <symlink-location>project-relative <tag>@Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean>%Jamfile<C:\Code\OTC\otc4-grpc-server\vcpkg_installed\vcpkg\blds\boost-thread\src\ost-1.77.0-7bf4191b6c.clean>.tag <target-os>windows <testing.execute>on <threadapi>win32 <threading>multi <toolset-msvc:version>144 <toolset>msvc <user-interface>console <variant>debug <vectorize>off <warnings-as-errors>off <warnings>all <windows-api>desktop

error: Unable to find file or target named

error:     '/boost//C:/Code/OTC/otc4-grpc-server/vcpkg_installed/x64-windows-static/debug/lib/boost_chrono-vc140-mt-gd.lib'

error: referred to from project at

error:     '..'



ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
  "name": "liborangeip",
  "version": "0.1.0",
  "description": [
    "Image processing library for project orange"
  ],
  "dependencies": [
    "boost-log",
    "boost-test",
    "numcpp",
    "gtest",
    "protobuf",
    "grpc",
    "jsoncpp",
    "openssl",
    "libzip",
    "opencv",
    "minizip-ng"
  ],
  "builtin-baseline": "24c6283930aee500f9ae584ebeaf3dc41d02bcd4",
  "overrides": [
    {
      "name": "numcpp",
      "version": "2.6.0"
    }
  ]
}

```
</details>

**Comment by dmiller423** [comment]
seconded, using clang for build is only difference 

It's also ridiculous VS has overridden my VCPKG_ROOT which downloads and build absolutely everything bc some..one thought it was a good idea - now I have a broken mess instead - wonderful, truly

**Comment by JonLiu1993** [comment]
@gideonr1, Thanks for posting issue, the latest boost-thread version is 1.85.0, boost-thread can be installed successfully, the version you are using is 1.77.0, do you have any specific requirements?
````
PS F:\Feature-test\test> ./vcpkg install boost-thread:x64-windows-static --editable
Computing installation plan...
The following packages are already installed:
    boost-thread:x64-windows-static@1.85.0#1
boost-thread:x64-windows-static is already installed
Total install time: 210 us
The package boost-thread is compatible with built-in CMake targets of FindBoost.cmake:

    find_package(Boost REQUIRED COMPONENTS thread)
    target_link_libraries(main PRIVATE Boost::thread)

or the generated cmake configs via:

    find_package(boost_thread REQUIRED CONFIG)
    target_link_libraries(main PRIVATE Boost::thread)

PS F:\Feature-test\test> ./vcpkg --version
vcpkg package management program version 2024-06-10-02590c430e4ed9215d27870138c2e579cc338772
````

**Comment by magnesj** [comment]
I can see the same error using GitHubAction with latest release of vcpkg 2024.07.12. Here Ninja is used, and only the release version is build. 

I am not able to reproduce on local machine.

https://github.com/magnesj/ResInsight/actions/runs/10388006261/job/28806974070#step:23:394

**Comment by magnesj** [comment]
We use `vcpkg` as a submodule. I updated the `vcpkg` submodule to `2024.07.12,` and I got the error situation described above. This operation will not use the latest available code in `vcpkg` submodule, the `baseline` in `vcpkg-configuration.json` is also required to be updated.

This operation is a bit hidden and difficult to find in the documentation.

https://github.com/OPM/ResInsight/pull/11614/files

I found the commit to use from here https://github.com/microsoft/vcpkg/tags
